src/node/base.js

```diff
@@ -0,0 +1,15 @@
+const ethUtil = require('ethereumjs-util')
+
+module.exports = class Node {
+  value () {
+    return this._value
+  }
+
+  setValue (v) {
+    this._value = v
+  }
+
+  hash () {
+    return ethUtil.keccak256(this.serialize())
```

Comment:
This might be a dumb question, but isnt it weird that `serialize` is only defined in the outer "inheriting" objects? If im following correctly, probably best to move serialize in here and then, if necessary, override it in the NullNode class

Reply:
Yeah, I think what you say makes sense. Will make sure to add at least a bare-bones `serialieze` to `Node`.

There are probably quite a few more of these points that need to be improved, the design is still not stable.